### ISSUE TITLE
Multiple numbers not displayed

### ISSUE THREAD

**Issue by TheLion** [sorvani/freepbx-helper-scripts#22]
Hi,

The output of the script (cm_to_yl_ab.php) only output a single number, not the other numbers that are entered in the contact in FreePBX.

I'm not looking in the phone, but in the XML in my browser.

I'm probably missing something, but not sure what.

**Comment by frest-de** [comment]
Yes, this does not work. I checked the SQL query, it does return only one number per contact.

**Comment by sorvani** [comment]
Already fixed — closing.

99804ef ("Read all Numbers") rewrote `cm_to_yl_ab.php` to group rows by display name and emit every number under a single `<DirectoryEntry>`. Verified on a live box: a group with two contacts, each having a Work and a Mobile number, returns all four.

One correction for anyone finding this later, because the diagnosis in this thread is wrong and will send you looking in the wrong place.

@frest-de — the query was never the problem. It returns **one row per number** and always did:

```sql
FROM contactmanager_group_entries AS cge
LEFT JOIN contactmanager_entry_numbers AS cen ON cen.entryid = cge.id
```

A contact with three numbers produces three rows. You can confirm it straight from the CLI with `mysql asterisk`.

The bug was in the output loop. It walked that flat result set tracking `$previousname`, and the `<Telephone>` echo was *inside* the name-change branch:

```php
foreach ($contacts as $contact) {
    if ($contact['displayname'] != $previousname) {
        echo "    <DirectoryEntry>\n";
        echo "        <Name>" . $contact['displayname'] . "</Name>\n";
        echo "        <Telephone ...>";   // <-- only ever runs on the first row of a contact
        echo "    </DirectoryEntry>\n";
        $previousname = $contact['displayname'];
    }
}
```

Every row after a contact's first fell through the `if` and was silently dropped. The rows were all there; nothing ever printed them.

`cm_to_fv_ab.php` and `cm_to_digum_aseries.php` never had this — their number echo is outside the name-change branch, which is why the same query behaves correctly there.

On **FreePBX 17**, use https://github.com/sorvani/freepbx17-phonebooks instead; see #25.
